fs2-aws/src/main/scala/fs2/aws/internal/KinesisProducerClient.scala

```diff
@@ -0,0 +1,37 @@
+package fs2.aws.internal
+
+import java.nio.ByteBuffer
+
+import cats.effect.Effect
+import com.amazonaws.auth.{AWSCredentialsProviderChain, DefaultAWSCredentialsProviderChain}
+import com.amazonaws.services.kinesis.producer.{
+  KinesisProducer,
+  KinesisProducerConfiguration,
+  UserRecordResult
+}
+import com.google.common.util.concurrent.ListenableFuture
+
+trait KinesisProducerClient[F[_]] {
+  def putData(streamName: String, partitionKey: String, data: ByteBuffer)(
+      implicit F: Effect[F]): F[ListenableFuture[UserRecordResult]]
+}
+
+class KinesisProducerClientImpl[F[_]] extends KinesisProducerClient[F] {
+
+  val credentials: AWSCredentialsProviderChain = new DefaultAWSCredentialsProviderChain()
```

Comment:
Think from a library perspective you should support different provider chains.  As I imagine this is the one that pulls from the machine and some folks could be using the inline or custom ones.

Reply:
The default provider chain searches through a bunch of different sources (ENV variables, launch arguments, file, and then EC2 metadata) so out of the box provides a bunch of different options for users. By exposing the KinesisProducerClient trait as well we offer users the flexibility to create their own implementation and supply their own credentials. 

https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html